zcash_client_backend/src/data_api.rs

```diff
@@ -513,16 +516,47 @@ impl<AccountId: Eq + Hash> WalletSummary<AccountId> {
         self.fully_scanned_height
     }
 
-    /// Returns the progress of scanning shielded outputs, in terms of the ratio between notes
-    /// scanned and the total number of notes added to the chain since the wallet birthday.
+    /// Returns the progress of scanning the chain to bring the wallet up to date.
     ///
-    /// This ratio should only be used to compute progress percentages, and the numerator and
-    /// denominator should not be treated as authoritative note counts. Returns `None` if the
-    /// wallet is unable to determine the size of the note commitment tree.
+    /// This progress metric is intended as an indicator of how close the wallet is to
+    /// general usability, including the ability to spend existing funds that were
+    /// previously spendable.
+    ///
+    /// The window over which progress is computed spans from the wallet's recovery height
+    /// to the current chain tip. This may be adjusted in future updates to better match
+    /// the intended semantics.
```

Comment:
If we compute from max-fully-scanned height (presuming fully-recovered), then because we use linear scanning, the scan progress is _always_ 0% (as its start height keeps moving) until scan is complete, and then it jumps to 100%.

We need the boundary between "recovery" and "scanning" to be semi-persistent - it needs to be fixed across multiple calls to `get_wallet_summary` in the short term, while ideally moving towards the chain tip over time so that the scan progress remains in a useful percentage range.

For this PR, I'm just using the recover-until height as a proxy for this in order to get the initial split in place, and we can decide on an improved approach later.